RecoParticleFlow/PFClusterTools/src/LinkByRecHit.cc

```diff
@@ -360,6 +412,294 @@ double LinkByRecHit::testTrackAndClusterByRecHit(const reco::PFRecTrack& track,
+
+        // Is this check really useful ?
+        if (std::abs(track_Z) > 700.25)
+          return -1.;
+      }
+      break;
+
+    case PFLayer::PS1:
+      [[fallthrough]];
+    case PFLayer::PS2:
+      //Note Alex: Nothing implemented for the
+      //PreShower (No resolution maps yet)
+      return -1.;
+    default:
+      return -1.;
+  }
+
+  // Check that, if the cluster is in the endcap,
+  // 0) the track indeed points to the endcap at vertex (DISABLED)
+  // 1) the track extrapolation is in the endcap too !
+  // 2) the track is in the same end-cap !
+  // PJ - 10-May-09
+  if (!barrel) {
+    // if ( fabs(trackEta) < 1.0 ) return -1;
+    if (!hcal && std::abs(track_Z) < 300.)
+      return -1.;
+    if (track_Z * clusterZ < 0.)
+      return -1.;
+  }
+  // Check that, if the cluster is in the barrel,
+  // 1) the track is in the barrel too !
+  if (barrel) {
+    if (!hcal && std::abs(track_Z) > 300.)
+      return -1.;
+  }
+
+  double dist = LinkByRecHit::computeDist(clustereta, clusterphi, tracketa, trackphi);
+
+  //Testing if Track can be linked by rechit to a cluster.
+  //A cluster can be linked to a track if the extrapolated position
+  //of the track to the ECAL ShowerMax/HCAL entrance falls within
+  //the boundaries of any cell that belongs to this cluster.
+
+  const auto fracsNbr = cluster_table.get<pf::cluster::FracsNbr>(icluster);
+
+  bool linkedbyrechit = false;
+  //loop rechits
+  for (size_t irechit : cluster_rechits) {
+    double fraction = rechit_table.get<pf::rechit::Fraction>(irechit);
+    if (fraction < 1E-4)
+      continue;
+
+    if (barrel || hcal) {  // barrel case matching in eta/phi
+                           // (and HCAL endcap too!)
+
+      //rechit size determination
+      // blown up by 50% (HCAL) to 100% (ECAL) to include cracks & gaps
+      // also blown up to account for multiple scattering at low pt.
+      double rhsizeEta = std::abs(rechit_table.get<pf::rechit::Corner3eta>(irechit) -
+                                  rechit_table.get<pf::rechit::Corner1eta>(irechit));
+      double rhsizePhi = std::abs(rechit_table.get<pf::rechit::Corner3phi>(irechit) -
+                                  rechit_table.get<pf::rechit::Corner1phi>(irechit));
+      if (rhsizePhi > M_PI)
+        rhsizePhi = 2. * M_PI - rhsizePhi;
+      if (hcal) {
+        const double mult = horesolscale * (1.50 + 0.5 / fracsNbr);
+        rhsizeEta = rhsizeEta * mult + 0.2 * std::abs(dHEta);
+        rhsizePhi = rhsizePhi * mult + 0.2 * fabs(dHPhi);
+
+      } else {
+        const double mult = 2.00 + 1.0 / (fracsNbr * std::min(1., 0.5 * trackPt));
+        rhsizeEta *= mult;
+        rhsizePhi *= mult;
+      }
+
+      //distance track-rechit center
+      // const math::XYZPoint& posxyz
+      // = rechit_cluster.position();
+      double deta = fabs(rechit_table.get<pf::rechit::Eta>(irechit) - tracketa);
+      double dphi = fabs(rechit_table.get<pf::rechit::Phi>(irechit) - trackphi);
```

Comment:
`reco::deltaPhi`

Reply:
we kept here the old implementation with fabs explicitly to not change reco output. If one accepts a reco change already in this PR (rather than a follow-up), this can be implemented.

Reply:
seems like it does not change results (even trivially), so I will introduce it